### PR TITLE
ttysvr: init at 0.3.3

### PR DESCRIPTION
**Pull request by griffi-gh**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by griffi-gh** [comment]
(Builds but does not run, and i don't have time to troubleshoot it rn)

```
thread 'main' panicked at /build/ttysvr-0.3.0-vendor.tar.gz/bevy_winit/src/lib.rs:132:14:
Failed to build event loop: NotSupported(NotSupportedError)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by griffi-gh** [comment]
0.3.2 removed WinitPlugin. 
Should work after an update.

**Comment by griffi-gh** [comment]
```
thread 'main' panicked at /build/ttysvr-0.3.3-vendor.tar.gz/bevy_render/src/renderer/mod.rs:186:10:
Unable to find a GPU! Make sure you have installed required drivers! For extra information, see: https://github.com/bevyengine/bevy/blob/latest/docs/linux_dependencies.md
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

File: pkgs/by-name/tt/ttysvr/package.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    repo = "ttysvr";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-qUwnx+hwd3PRzz1TlQzPAppj6aTZZpTG2e5cBaA3zZI=";
+  };
+  cargoHash = "sha256-kIE8+FUS3i8Ulkj35lO1iRmFx5x+r89THVuzNzAv6tE=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+  buildInputs = [
```
Comment (GaetanLepage) [inline pkgs/by-name/tt/ttysvr/package.nix:29]:
```suggestion
  ];

  buildInputs = [
```

File: pkgs/by-name/tt/ttysvr/package.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    patchelf $out/bin/ttysvr \
+      --add-rpath ${
+        lib.makeLibraryPath [
+          libGL
+          vulkan-loader
+        ]
+      }
+  '';
+
+  meta = {
```
Comment (GaetanLepage) [inline pkgs/by-name/tt/ttysvr/package.nix:47]:
```suggestion
  '';

  nativeInstallCheckInputs = [
    versionCheckHook
  ];
  versionCheckProgramArg = [ "--version" ];
  doInstallCheck = true;

  passthru = {
    updateScript = nix-update-script { };
  };

  meta = {
```

File: pkgs/by-name/tt/ttysvr/package.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    description = "Screen saver for your terminal";
+    homepage = "https://github.com/cxreiff/ttysvr";
+    changelog = "https://github.com/cxreiff/ttysvr/releases";
```
Comment (GaetanLepage) [inline pkgs/by-name/tt/ttysvr/package.nix:50]:
```suggestion
    changelog = "https://github.com/cxreiff/ttysvr/releases/tag/v${version}";
```

**Review by GaetanLepage — APPROVED** [review]

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347166`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ttysvr</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ttysvr</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Pull request merged**